realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTest.java

```diff
@@ -1168,7 +1171,7 @@ public void run() {
                 realm.addChangeListener(new RealmChangeListener() {
                     @Override
                     public void onChange() {
-                        if (globalCommitInvocations.incrementAndGet() == 2) {
+                        if (realm.where(Dog.class).count() == 2) {
```

Comment:
The notification behaviour will be changed when multi-processes support merged. The notification won't be guaranteed to be triggered on every time the transaction commited, instead, it will be guaranteed to be triggered when the latest transaction committed.